server/journeys/referral/details/schemas.ts

```diff
@@ -61,4 +53,16 @@ export const schemaFactory =
         ...val,
         incidentTime: val.hour?.success && val.minute?.success ? `${val.hour.data}:${val.minute.data}` : null,
       }))
+      .and(
+        z.object({
+          incidentLocation: z
+            .string({ message: INCIDENT_LOCATION_MSG })
+            .refine(val => incidentLocationMap.has(val), INCIDENT_LOCATION_MSG)
+            .transform(val => incidentLocationMap.get(val)),
+          incidentType: z
+            .string({ message: INCIDENT_TYPE_MSG })
+            .refine(val => incidentTypeMap.has(val), INCIDENT_TYPE_MSG)
+            .transform(val => incidentTypeMap.get(val)),
+        }),
+      )
```

Comment:
.and() cannot be followed by `.strict()`,
and `merge` `union` `intersection` doesn't play nicely with inferring schemaType.

hence in PR #108  i'm adding a util class so that we can have all req.body properties defined in a single object, and yet the object.refine().transform() functions will always be executed.

Reply:
Which class or code is that?

Reply:
I understand this all better now - I'll wait for #108